packages/service-library/src/servicelib/fastapi/long_running_tasks/_server.py

```diff
@@ -40,13 +40,14 @@ async def on_startup() -> None:
             stale_task_detect_timeout_s=stale_task_detect_timeout_s,
         )
 
-        # add error handlers
-        app.add_exception_handler(BaseLongRunningError, base_long_running_error_handler)
-
     async def on_shutdown() -> None:
         if app.state.long_running_task_manager:
             task_manager: TasksManager = app.state.long_running_task_manager
             await task_manager.close()
 
     app.add_event_handler("startup", on_startup)
     app.add_event_handler("shutdown", on_shutdown)
+
+    # add error handlers
+    # NOTE: Exception handler can not be added during the on_startup script, otherwise not working correctly
+    app.add_exception_handler(BaseLongRunningError, base_long_running_error_handler)
```

Comment:
👍 and I get it you checked we did it correctly in the services right?

Reply:
Yes correct 👍 this was the only place we have it in the startup event function